src/harness/util.py

```diff
@@ -279,3 +279,129 @@ def convertRequestToRequestWithCpiSignature(private_key, cpi_id,
+    mL = np.floor((minlon - iLon)/arcsec)
+    mH = np.ceil((maxlon - iLon)/arcsec)
+        
+    latgrid = np.arange(iLat + nL*arcsec, iLat + nH *arcsec, arcsec)
+    longrid = np.arange(iLon + mL * arcsec, iLon + mH *arcsec, arcsec)
+    
+    if len(latgrid) > 0 and len(latgrid)>0:
+        ppa = []
+
+        for lat in latgrid:
+            for lon in longrid:
+                ppa_check_point = Point(lon, lat)
+                if ppa_check_point.within(poly):
+                        ppa.extend([[lon, lat]])
+
+        if (len(ppa) == 1):
+            return 0, ppa[0][0], ppa[0][1]
+        else:
+            return -1, -1, -1
+    else:
+        return -1, -1, -1
+
+def QueryPropagationAntennaModel(requestJson):
+    import os
+    from reference_models.propagation import wf_itm
+    from reference_models.propagation import wf_hybrid
+    from reference_models.antenna import antenna
+    from reference_models.geo import vincenty
+    from reference_models.geo import nlcd
+    TERRAIN_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'ned')
+    NLCD_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'nlcd')
+    ITU_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'itu')
+    wf_itm.ConfigureTerrainDriver(terrain_dir=TERRAIN_TEST_DIR)
+    wf_itm.ConfigureItuDrivers(itu_dir=ITU_TEST_DIR)
+    wf_hybrid.ConfigureTerrainDriver(terrain_dir=TERRAIN_TEST_DIR)
+    wf_hybrid.wf_itm.ConfigureItuDrivers(itu_dir=ITU_TEST_DIR)
+    nlcd_driver = nlcd.NlcdDriver(NLCD_TEST_DIR)
+
+    request = json.loads(requestJson)
+    reliabilityLevel = request['reliabilityLevel']
+    if reliabilityLevel not in [-1, 0.05, 0.95]:
+        response = 404
+        return response
+    Tx = request['cbsd']
+    if ('fss' in request) and ('ppa' in request):
+        response = 404
+        return response
+    elif 'ppa' in request:
+        Rx ={}
+        Rx['height'] = 1.5
+        isfss = False
+        coordinates = [];
+        ppa = request['ppa']
+        boundarysPoints = ppa['geometry']['coordinates']
+        boundarypoints = []
+        for BP in boundarysPoints:
+            boundarypoints.append([BP[1], BP[0]])
+        
+        regionVal = nlcd_driver.RegionNlcdVote(boundarypoints)
+        ARCSEC = 2
+        res, lat, lon = getPPinPPA(boundarysPoints, ARCSEC)
+        if res== 0:
+            Rx['latitude'] = lat
+            Rx['longitude']= lon
+        else:
+            response = 404
+            return response
+
+    elif 'fss' in request:
+        isfss = True
+        Rx = request['fss']
+    else:
+        response = 404
+        return response
+    
+# ITM pathloss (if receiver type is FSS) or the hybrid model pathloss (if receiver type is PPA) and corresponding antenna gains.
+# The test specification notes that the SAS UUT shall use default values for w1 and w2 in the ITM model.
+    Result = {}
```

Comment:
Best to be consistent the way you name variable. Sometimes you use an upper case for first char, sometimes not.
I think that elsewhere  in SAS code, we have used consistently the "all lower_case with underscore format"  (example:  config_parser) for variable names.